pkg/ingester/shipper.go

```diff
@@ -162,6 +168,8 @@ func (s *Shipper) Sync(ctx context.Context) (uploaded int, err error) {
 			level.Error(s.logger).Log("msg", "shipping failed", "block", m.ULID, "err", err)
 			uploadErrs++
 			continue
+		} else {
+			s.metrics.lastSuccessfulUploadTime.WithLabelValues(s.bucket.Name()).SetToCurrentTime()
 		}
```

Comment:
Nit: I'd drop `else {` part here, as subsequent lines also handle the case when upload succeeded, and they should be all together. (I'd also move `s.metrics.lastSuccessfulUploadTime` update after `s.metrics.uploads.Inc()`)

Reply:
It would be nice to add test to check that new metric is updated on successful upload.